extensions/azurecore/src/azureResource/utils.ts

```diff
@@ -284,3 +286,73 @@ export async function getSelectedSubscriptions(appContext: AppContext, account?:
 	}
 	return result;
 }
+
+export async function makeGetRequest(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, ignoreErrors: boolean = false, url: string): Promise<GetRequestResult> {
+	const result: GetRequestResult = { response: {}, errors: [] };
+
+	if (!account?.properties?.tenants || !Array.isArray(account.properties.tenants)) {
+		const error = new Error(localize('azure.accounts.getSubscriptions.invalidParamsError', "Invalid account"));
+		if (!ignoreErrors) {
+			throw error;
+		}
```

Comment:
Make a function like below:
/**
* check specified 'condition'. Does nothing if the condition if false
* else throws the 'errorFailed' error unless ignoreError is true in which case errorFailed is pushed at the end of the 'errors' array.
*/
errorUnless(condition: any, errorIfFailed: Error, ignoreError?: boolean, errors: Error[]) {
// implementation here.
}

Then each of these statements becomes:
errorUnless(
    !account?.properties?.tenants || !Array.isArray(account.properties.tenants, 
    new Error(localize('azure.accounts.getSubscriptions.invalidParamsError', "Invalid account")),
    ignoreErrors,
    errors
);